app/models/user_api.rb

```diff
@@ -0,0 +1,28 @@
+class UserApi < ApplicationRecord
+  def overwrite_key
+    update(key: UserApi.generate_key)
+  end
+
+  def self.save_key(params, uid)
```

Comment:
Don't think we need this method. If you setup a relationship between this model and its user you can do `user.user_api.update(user_api_params)`. See my note regarding using `uid` as a column name instead of `user_id`. Once you fix that you should be able to set up a `belongs_to :user` in here.